ragna/core/_rag.py

```diff
@@ -195,6 +195,9 @@ async def prepare(self) -> Message:
         await self._run(self.source_storage.store, self.documents)
         self._prepared = True
 
+        system_prompt = self.assistant._make_system_content()
```

Comment:
It feels easiest to include the basic system prompt here, but maybe it should be regenerated and prepended to the list of messages for every call to the LLM and not stored here. This would make the most sense if the Assistant might truncate the message list and chop off the system prompt.

Reply:
I don't understand why the chat would need the system prompt of the assistant. Instead of creating it here and pass it to the Assistant as part of the messages, why not let the Assistant create it in the first place?

Reply:
I think I agree. I also mostly talked myself out of putting it here as well. I think I wanted `_messages` to be a complete historical list of messages that could be sent to the LLM, but system prompts can stay hard coded on the Assistant (or better yet, in a config somewhere?)

Reply:
Just be sure there's no attempt at a 'universal' system prompt anywhere - on the assistant we'll have a general one for .answer(), but e.g. we'll want different system prompts for things like all the pre-process steps or anything that's outside the standard chain. Simply to say, it's reasonable to have a standard system prompt on .answer() but we don't want it fixed on the general llm calls in .generate() (wip).